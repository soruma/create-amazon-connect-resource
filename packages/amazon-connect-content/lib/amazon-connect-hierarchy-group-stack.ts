import { readFileSync } from 'node:fs';
import * as path from 'node:path';
import * as cdk from 'aws-cdk-lib';
import * as connect from 'aws-cdk-lib/aws-connect';
import { Construct } from 'constructs';
import { parse as JSONCParse } from 'jsonc-parser';

interface HierarchyGroup {
  name: string;
  children?: HierarchyGroup[];
}

/**
 * This stack creates user hierarchy groups for Amazon Connect.
 */
interface AmazonConnectHierarchyGroupProps {
  /**
   * Amazon Connect instance ARN
   */
  connectInstanceArn: string;
}

export class AmazonConnectHierarchyGroupStack extends cdk.NestedStack {
  private props: AmazonConnectHierarchyGroupProps;

  constructor(scope: Construct, id: string, props: AmazonConnectHierarchyGroupProps) {
    super(scope, id);

    this.props = props;

    const data = readFileSync(path.join(__dirname, '..', 'config', 'user-hierarchy-groups.jsonc'), {
      encoding: 'utf8',
    });
    const config = JSONCParse(data) as HierarchyGroup[];

    config.forEach((group, idx) => {
      this.createHierarchyGroup(group, undefined, `Root${idx}`);
    });
  }

  private createHierarchyGroup(group: HierarchyGroup, parentGroupArn: string | undefined, uniqueId: string) {
    const groupId = `${uniqueId}_${group.name.replace(/\s+/g, '')}`;

    const hierarchyGroup = new connect.CfnUserHierarchyGroup(this, groupId, {
      instanceArn: this.props.connectInstanceArn,
      name: group.name,
      parentGroupArn: parentGroupArn,
    });

    if (group.children) {
      group.children.forEach((child, idx) => {
        this.createHierarchyGroup(child, hierarchyGroup.attrUserHierarchyGroupArn, `${groupId}_${idx}`);
      });
    }
  }
}
