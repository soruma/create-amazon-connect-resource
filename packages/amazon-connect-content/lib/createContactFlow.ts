import { readFileSync } from 'node:fs';
import * as path from 'node:path';
import { CfnContactFlow, CfnContactFlowModule } from 'aws-cdk-lib/aws-connect';
import { Construct } from 'constructs';
import { parse as JSONCParse } from 'jsonc-parser';

/**
 * Configuration for contact flow
 */
export interface ContactFlowConfig {
  /**
   * Contact flow id
   */
  readonly id: string;

  readonly file: string;
}

/**
 * Configuration for contact flows
 */
export interface ContactFlowsConfig {
  /**
   * Configuration for module contact flows
   */
  readonly modules: ContactFlowConfig[];

  /**
   * Configuration for contact flows
   */
  readonly contactFlows: ContactFlowConfig[];
}

/**
 * Creates contact flow configurations based on the provided configuration
 * @param {Construct} scope - The CDK Construct scope
 * @param {string} connectInstanceArn - The Amazon Connect instance ARN
 * @returns {ContactFlowsConfig} - The parsed contact flow configuration
 */
export const createContactFlow = (scope: Construct, connectInstanceArn: string): ContactFlowsConfig => {
  const data = readFileSync(path.join(__dirname, '..', 'config', 'contact-flows.jsonc'), { encoding: 'utf8' });
  const config = JSONCParse(data) as ContactFlowsConfig;

  // Add modules
  for (const module of config.modules) {
    const contactFlow = readFileSync(path.join(__dirname, '..', 'config', 'modules', module.file), {
      encoding: 'utf8',
    });
    new CfnContactFlowModule(scope, `ContactFlowModule${module.id}`, {
      instanceArn: connectInstanceArn,
      name: contactFlow.Metadata.name,
      description: contactFlow.Metadata.description,
      content: contactFlow,
    });
  }

  // Add contact flow
  for (const contactFlows of config.contactFlows) {
    const contactFlow = readFileSync(path.join(__dirname, '..', 'config', 'modules', contactFlows.file), {
      encoding: 'utf8',
    });
    new CfnContactFlow(scope, `ContactFlowModule${contactFlows.id}`, {
      instanceArn: connectInstanceArn,
      name: contactFlow.Metadata.name,
      description: contactFlow.Metadata.description,
      type: contactFlow.Metadata.type,
      content: contactFlow,
    });
  }

  return config;
};
