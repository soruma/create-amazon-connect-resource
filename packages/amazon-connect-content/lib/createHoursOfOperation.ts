import { readFileSync } from 'node:fs';
import * as path from 'node:path';
import { CfnHoursOfOperation } from 'aws-cdk-lib/aws-connect';
import { Construct } from 'constructs';
import { parse as JSONCParse } from 'jsonc-parser';

/**
 * Configuration for hours of operation
 */
export interface HoursOfOperationConfig {
  /**
   * Hours of operation id
   */
  readonly id: string;

  /**
   * Hours of operation name
   */
  readonly name: string;

  /**
   * Hours of operation description
   */
  readonly description: string;

  /**
   * Hours of operation timezone
   */
  readonly timeZone: string;

  /**
   * Hours of operation configuration
   */
  readonly config: CfnHoursOfOperation.HoursOfOperationConfigProperty[];

  /**
   * Amazon Connect hours of operation ARN
   */
  hoursOfOperationArn: string;
}

/**
 * Creates hours of operation resources based on the provided configuration
 * @param {Construct} scope - The CDK Construct scope
 * @param {string} connectInstanceArn - The Amazon Connect instance ARN
 * @returns {HoursOfOperationConfig[]} An array of hours of operation configurations
 */
export const createHoursOfOperation = (scope: Construct, connectInstanceArn: string): HoursOfOperationConfig[] => {
  const data = readFileSync(path.join(__dirname, '..', 'config', 'hours-of-operations.jsonc'), { encoding: 'utf8' });
  const configs = JSONCParse(data) as HoursOfOperationConfig[];

  for (const config of configs) {
    const hoursOfOperation = new CfnHoursOfOperation(scope, `HoursOfOperation${config.id}`, {
      instanceArn: connectInstanceArn,
      ...config,
    });
    config.hoursOfOperationArn = hoursOfOperation.attrHoursOfOperationArn;
  }

  return configs;
};
