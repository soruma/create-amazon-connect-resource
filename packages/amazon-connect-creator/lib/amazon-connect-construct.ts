import { readFileSync } from 'node:fs';
import * as path from 'node:path';
import { Construct } from 'constructs';

import * as connect from 'aws-cdk-lib/aws-connect';
import * as iam from 'aws-cdk-lib/aws-iam';
import * as kms from 'aws-cdk-lib/aws-kms';
import * as s3 from 'aws-cdk-lib/aws-s3';

export type IdentityManagementType = 'SAML' | 'CONNECT_MANAGED' | 'EXISTING_DIRECTORY';

interface AmazonConnectConstructProps {
  /**
   * Amazon Connect instance alias
   */
  connectInstanceAlias: string;

  /**
   * Mandatory element which enables inbound calls on new instance.
   */
  inboundCalls: boolean;

  /**
   * Mandatory element which enables outbound calls on new instance.
   */
  outboundCalls: boolean;

  /**
   * Boolean flag which enables CONTACTFLOW_LOGS on an instance.
   */
  contactflowLogs: boolean;

  /**
   * Boolean flag which enables AUTO_RESOLVE_BEST_VOICES on an instance.
   */
  autoResolveBestVoices: boolean;

  /**
   * The identity management type.
   */
  identityManagementType: IdentityManagementType;

  /**
   * The identifier for the directory.
   */
  directoryId?: string;

  /**
   * Amazon connect data storage
   */
  dataStorageBucket?: s3.Bucket;

  /**
   * business hours time zone
   */
  businessHoursTimeZone: string;
}

export class AmazonConnectConstruct extends Construct {
  private readonly props: AmazonConnectConstructProps;
  readonly connectInstance: connect.CfnInstance;

  constructor(scope: Construct, id: string, props: AmazonConnectConstructProps) {
    super(scope, id);

    this.props = props;

    this.connectInstance = new connect.CfnInstance(this, id, {
      attributes: {
        inboundCalls: this.props.inboundCalls,
        outboundCalls: this.props.outboundCalls,
        contactflowLogs: this.props.contactflowLogs,
        autoResolveBestVoices: this.props.autoResolveBestVoices,
      },
      identityManagementType: this.props.identityManagementType,
      instanceAlias: this.props.connectInstanceAlias,
      directoryId: this.props.directoryId,
    });

    if (this.props.dataStorageBucket !== undefined) {
      this.props.dataStorageBucket.grantWrite(new iam.ServicePrincipal('connect.amazonaws.com'));

      this.createInstanceStorageConfig();
    }

    this.createBusinessHours();
  }

  createInstanceStorageConfig() {
    if (this.props.dataStorageBucket === undefined) return;

    const encriptionKey = kms.Key.fromLookup(this, 'EncryptKeyLookup', {
      aliasName: 'alias/aws/connect',
    });

    new connect.CfnInstanceStorageConfig(this, 'CallRecordingsStorageConfig', {
      instanceArn: this.connectInstance.attrArn,
      resourceType: 'CALL_RECORDINGS',
      storageType: 'S3',
      s3Config: {
        bucketName: this.props.dataStorageBucket.bucketName,
        bucketPrefix: 'CallRecordings',
        encryptionConfig: {
          encryptionType: 'KMS',
          keyId: encriptionKey.keyArn,
        },
      },
    });

    new connect.CfnInstanceStorageConfig(this, 'ChatTranscriptsStorageConfig', {
      instanceArn: this.connectInstance.attrArn,
      resourceType: 'CHAT_TRANSCRIPTS',
      storageType: 'S3',
      s3Config: {
        bucketName: this.props.dataStorageBucket.bucketName,
        bucketPrefix: 'ChatTranscripts',
        encryptionConfig: {
          encryptionType: 'KMS',
          keyId: encriptionKey.keyArn,
        },
      },
    });
  }

  createBusinessHours(): connect.CfnHoursOfOperation {
    const data = readFileSync(path.join(__dirname, '..', 'config', 'business_hours.json'), { encoding: 'utf8' });
    const config = JSON.parse(data) as connect.CfnHoursOfOperation.HoursOfOperationConfigProperty[];

    return new connect.CfnHoursOfOperation(this, 'HoursOfOperation', {
      instanceArn: this.connectInstance.attrArn,
      name: 'Business Hour',
      config,
      timeZone: this.props.businessHoursTimeZone,
    });
  }
}
