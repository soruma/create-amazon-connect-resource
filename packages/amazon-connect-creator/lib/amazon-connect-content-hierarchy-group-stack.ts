import * as cdk from 'aws-cdk-lib';

import * as connect from 'aws-cdk-lib/aws-connect';
import { Construct } from 'constructs';

interface AmazonConnectContentHierarchyGroupProps {
  /**
   * Amazon Connect instance ARN
   */
  connectInstanceArn: string;
}

export class AmazonConnectContentHierarchyGroupStack extends cdk.NestedStack {
  private props: AmazonConnectContentHierarchyGroupProps;

  constructor(scope: Construct, id: string, props: AmazonConnectContentHierarchyGroupProps) {
    super(scope, id);

    this.props = props;

    const salesDepartment = new connect.CfnUserHierarchyGroup(this, 'SalesDepartmentContactUserHierarchyGroup', {
      instanceArn: this.props.connectInstanceArn,
      name: 'Sales department',
    });

    new connect.CfnUserHierarchyGroup(this, 'SalesDepartmentTeam1ContactUserHierarchyGroup', {
      instanceArn: this.props.connectInstanceArn,
      name: 'Team 1',
      parentGroupArn: salesDepartment.attrUserHierarchyGroupArn,
    });

    new connect.CfnUserHierarchyGroup(this, 'SalesDepartmentTeam2ContactUserHierarchyGroup', {
      instanceArn: this.props.connectInstanceArn,
      name: 'Team 2',
      parentGroupArn: salesDepartment.attrUserHierarchyGroupArn,
    });

    const supportDepartment = new connect.CfnUserHierarchyGroup(this, 'SupportDepartmentContactUserHierarchyGroup', {
      instanceArn: this.props.connectInstanceArn,
      name: 'Support department',
    });

    new connect.CfnUserHierarchyGroup(this, 'SupportDepartmentTeam1ContactUserHierarchyGroup', {
      instanceArn: this.props.connectInstanceArn,
      name: 'Team 1',
      parentGroupArn: supportDepartment.attrUserHierarchyGroupArn,
    });
  }
}
