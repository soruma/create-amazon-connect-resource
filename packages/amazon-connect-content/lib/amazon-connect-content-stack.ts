import { readFileSync } from 'node:fs';
import * as path from 'node:path';
import * as cdk from 'aws-cdk-lib';
import * as connect from 'aws-cdk-lib/aws-connect';
import { Construct } from 'constructs';
import { parse as JSONCParse } from 'jsonc-parser';

import { AmazonConnectHierarchyGroupStack } from './amazon-connect-hierarchy-group-stack';

interface AmazonConnectContentStackProps extends cdk.StackProps {
  /**
   * Amazon Connect instance ARN
   */
  connectInstanceArn: string;

  /**
   * Create sample Amazon Connect organization hierarchies
   */
  createHierarchy: boolean;

  /**
   * business hours time zone
   */
  businessHoursTimeZone: string;
}

export class AmazonConnectContentStack extends cdk.Stack {
  private readonly props: AmazonConnectContentStackProps;

  constructor(scope: Construct, id: string, props: AmazonConnectContentStackProps) {
    super(scope, id, props);

    this.props = props;

    this.createBusinessHours();

    if (this.props.createHierarchy) {
      this.createHierarchy();
    }
  }

  createBusinessHours(): connect.CfnHoursOfOperation {
    const data = readFileSync(path.join(__dirname, '..', 'config', 'business-hours.jsonc'), { encoding: 'utf8' });
    const config = JSONCParse(data) as connect.CfnHoursOfOperation.HoursOfOperationConfigProperty[];

    return new connect.CfnHoursOfOperation(this, 'HoursOfOperation', {
      instanceArn: this.props.connectInstanceArn,
      name: 'Business Hours',
      description: 'Standard business hours for operations',
      config,
      timeZone: this.props.businessHoursTimeZone,
    });
  }

  createHierarchy() {
    const data = readFileSync(path.join(__dirname, '..', 'config', 'hierarchy-structure.jsonc'), { encoding: 'utf8' });
    const userHierarchyStructure = JSONCParse(data) as connect.CfnUserHierarchyStructure.UserHierarchyStructureProperty;

    new connect.CfnUserHierarchyStructure(this, 'DepartmentUserHierarchyStructure', {
      instanceArn: this.props.connectInstanceArn,
      userHierarchyStructure,
    });

    new AmazonConnectHierarchyGroupStack(this, 'AmazonConnectHierarchyGroupStack', {
      connectInstanceArn: this.props.connectInstanceArn,
    });
  }
}
