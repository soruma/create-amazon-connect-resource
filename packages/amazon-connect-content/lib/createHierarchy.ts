import { readFileSync } from 'node:fs';
import * as path from 'node:path';
import { CfnUserHierarchyStructure } from 'aws-cdk-lib/aws-connect';
import { Construct } from 'constructs';
import { parse as JSONCParse } from 'jsonc-parser';

import { AmazonConnectHierarchyGroupStack } from './amazon-connect-hierarchy-group-stack';

/**
 * Creates a hierarchy group in Amazon Connect.
 * @param {Construct} scope - The CDK Construct scope
 * @param {string} connectInstanceArn - The Amazon Connect instance ARN
 */
export const createHierarchy = (scope: Construct, connectInstanceArn: string) => {
  const data = readFileSync(path.join(__dirname, '..', 'config', 'hierarchy-structure.jsonc'), { encoding: 'utf8' });
  const userHierarchyStructure = JSONCParse(data) as CfnUserHierarchyStructure.UserHierarchyStructureProperty;

  new CfnUserHierarchyStructure(scope, 'DepartmentUserHierarchyStructure', {
    instanceArn: connectInstanceArn,
    userHierarchyStructure,
  });

  new AmazonConnectHierarchyGroupStack(scope, 'AmazonConnectHierarchyGroupStack', {
    connectInstanceArn: connectInstanceArn,
  });
};
