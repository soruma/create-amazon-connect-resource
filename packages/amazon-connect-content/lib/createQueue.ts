import { readFileSync } from 'node:fs';
import * as path from 'node:path';
import { CfnQueue } from 'aws-cdk-lib/aws-connect';
import { Construct } from 'constructs';
import { parse as JSONCParse } from 'jsonc-parser';

import { HoursOfOperationConfig } from './createHoursOfOperation';

/**
 * Configuration for hours of operation
 */
export interface QueueConfig {
  /**
   * Queue id
   */
  readonly id: string;

  /**
   * Queue name
   */
  readonly name: string;

  /**
   * Queue description
   */
  readonly description: string;

  /**
   * Hours of operation ID
   */
  readonly hoursOfOperationId: string;

  /**
   * Queue ARN
   */
  queueArn: string;
}

/**
 * Creates queue configurations based on the provided configuration
 * @param {Construct} scope - The CDK Construct scope
 * @param {string} connectInstanceArn - The Amazon Connect instance ARN
 * @param {HoursOfOperationConfig[]} hoursOfOperationConfigs - The configuration for hours of operation
 * @returns {QueueConfig[]} An array of queue configurations
 */
export const createQueue = (
  scope: Construct,
  connectInstanceArn: string,
  hoursOfOperationConfigs: HoursOfOperationConfig[],
): QueueConfig[] => {
  const data = readFileSync(path.join(__dirname, '..', 'config', 'queues.jsonc'), { encoding: 'utf8' });
  const configs = JSONCParse(data) as QueueConfig[];

  for (const config of configs) {
    const hoursOfOperationConfig = hoursOfOperationConfigs.find((hoursOfOperation) => {
      return hoursOfOperation.id === config.hoursOfOperationId;
    });

    if (!hoursOfOperationConfig) {
      break;
    }
    const queue = new CfnQueue(scope, `Queue${config.id}`, {
      instanceArn: connectInstanceArn,
      hoursOfOperationArn: hoursOfOperationConfig.hoursOfOperationArn,
      ...config
    });
    config.queueArn = queue.attrQueueArn;
  }

  return configs;
};
