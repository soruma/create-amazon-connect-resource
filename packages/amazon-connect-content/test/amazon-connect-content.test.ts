import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { test as baseTest, describe, expect, it } from 'vitest';

import { AmazonConnectContentStack } from '../lib/amazon-connect-content-stack';
import { AmazonConnectHierarchyGroupStack } from '../lib/amazon-connect-hierarchy-group-stack';

const test = baseTest.extend<{
  app: cdk.App;
}>({
  // biome-ignore lint: lint/correctness/noEmptyPattern
  app: ({}, use) => use(new cdk.App()),
});

describe('AmazonConnectContentStack', () => {
  describe('Matches the snapshot', () => {
    test('Create hierarchy', ({ app }) => {
      const stack = new AmazonConnectContentStack(app, 'AmazonConnectContentStack-test', {
        env: {
          account: '000000000000',
          region: 'us-east-1',
        },
        connectInstanceArn: 'arn:aws:connect:us-east-1:000000000000:instance/00000000-0000-0000-0000-000000000000',
        createHierarchy: true,
      });

      const template = Template.fromStack(stack);
      expect(template.toJSON()).toMatchSnapshot();

      const hierarchyGroupStackNodes = stack.node.children.filter(
        (child) => child instanceof AmazonConnectHierarchyGroupStack,
      );

      for (const child of hierarchyGroupStackNodes) {
        expect(Template.fromStack(child as AmazonConnectHierarchyGroupStack).toJSON()).toMatchSnapshot();
      }
    });

    test('Not create hierarchy', () => {
      const app = new cdk.App();

      const stack = new AmazonConnectContentStack(app, 'AmazonConnectContentStack-test', {
        env: {
          account: '000000000000',
          region: 'us-east-1',
        },
        connectInstanceArn: 'arn:aws:connect:us-east-1:000000000000:instance/00000000-0000-0000-0000-000000000000',
        createHierarchy: false,
      });

      const template = Template.fromStack(stack);
      expect(template.toJSON()).toMatchSnapshot();
    });
  });
});
