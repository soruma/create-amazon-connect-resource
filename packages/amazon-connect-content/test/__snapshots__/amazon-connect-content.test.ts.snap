// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AmazonConnectContentStack > Matches the snapshot > Create hierarchy 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "AmazonConnectHierarchyGroupStackNestedStackAmazonConnectHierarchyGroupStackNestedStackResource3C4DFE15": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "TemplateURL": {
          "Fn::Join": [
            "",
            [
              "https://s3.us-east-1.",
              {
                "Ref": "AWS::URLSuffix",
              },
              "/cdk-hnb659fds-assets-000000000000-us-east-1/f4c7192531e2169c783120481af23169ff2567236f1d9551c00ccb6f2ba5bce0.json",
            ],
          ],
        },
      },
      "Type": "AWS::CloudFormation::Stack",
      "UpdateReplacePolicy": "Delete",
    },
    "DepartmentUserHierarchyStructure": {
      "Properties": {
        "InstanceArn": "arn:aws:connect:us-east-1:000000000000:instance/00000000-0000-0000-0000-000000000000",
        "UserHierarchyStructure": {
          "LevelOne": {
            "HierarchyLevelId": "division",
            "Name": "Division",
          },
          "LevelThree": {
            "HierarchyLevelId": "team",
            "Name": "Team",
          },
          "LevelTwo": {
            "HierarchyLevelId": "department",
            "Name": "Department",
          },
        },
      },
      "Type": "AWS::Connect::UserHierarchyStructure",
    },
    "HoursOfOperation": {
      "Properties": {
        "Config": [
          {
            "Day": "MONDAY",
            "EndTime": {
              "Hours": 17,
              "Minutes": 0,
            },
            "StartTime": {
              "Hours": 9,
              "Minutes": 0,
            },
          },
          {
            "Day": "TUESDAY",
            "EndTime": {
              "Hours": 17,
              "Minutes": 0,
            },
            "StartTime": {
              "Hours": 9,
              "Minutes": 0,
            },
          },
          {
            "Day": "WEDNESDAY",
            "EndTime": {
              "Hours": 17,
              "Minutes": 0,
            },
            "StartTime": {
              "Hours": 9,
              "Minutes": 0,
            },
          },
          {
            "Day": "THURSDAY",
            "EndTime": {
              "Hours": 17,
              "Minutes": 0,
            },
            "StartTime": {
              "Hours": 9,
              "Minutes": 0,
            },
          },
          {
            "Day": "FRIDAY",
            "EndTime": {
              "Hours": 17,
              "Minutes": 0,
            },
            "StartTime": {
              "Hours": 9,
              "Minutes": 0,
            },
          },
        ],
        "Description": "Standard business hours for operations",
        "InstanceArn": "arn:aws:connect:us-east-1:000000000000:instance/00000000-0000-0000-0000-000000000000",
        "Name": "Business Hours",
        "TimeZone": "UTC",
      },
      "Type": "AWS::Connect::HoursOfOperation",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;

exports[`AmazonConnectContentStack > Matches the snapshot > Create hierarchy 2`] = `
{
  "Resources": {
    "Root0SalesDivision": {
      "Properties": {
        "InstanceArn": "arn:aws:connect:us-east-1:000000000000:instance/00000000-0000-0000-0000-000000000000",
        "Name": "Sales Division",
      },
      "Type": "AWS::Connect::UserHierarchyGroup",
    },
    "Root0SalesDivision0DomesticDepartment": {
      "Properties": {
        "InstanceArn": "arn:aws:connect:us-east-1:000000000000:instance/00000000-0000-0000-0000-000000000000",
        "Name": "Domestic Department",
        "ParentGroupArn": {
          "Fn::GetAtt": [
            "Root0SalesDivision",
            "UserHierarchyGroupArn",
          ],
        },
      },
      "Type": "AWS::Connect::UserHierarchyGroup",
    },
    "Root0SalesDivision0DomesticDepartment0TeamA": {
      "Properties": {
        "InstanceArn": "arn:aws:connect:us-east-1:000000000000:instance/00000000-0000-0000-0000-000000000000",
        "Name": "Team A",
        "ParentGroupArn": {
          "Fn::GetAtt": [
            "Root0SalesDivision0DomesticDepartment",
            "UserHierarchyGroupArn",
          ],
        },
      },
      "Type": "AWS::Connect::UserHierarchyGroup",
    },
    "Root0SalesDivision0DomesticDepartment1TeamB": {
      "Properties": {
        "InstanceArn": "arn:aws:connect:us-east-1:000000000000:instance/00000000-0000-0000-0000-000000000000",
        "Name": "Team B",
        "ParentGroupArn": {
          "Fn::GetAtt": [
            "Root0SalesDivision0DomesticDepartment",
            "UserHierarchyGroupArn",
          ],
        },
      },
      "Type": "AWS::Connect::UserHierarchyGroup",
    },
    "Root0SalesDivision1InternationalDepartment": {
      "Properties": {
        "InstanceArn": "arn:aws:connect:us-east-1:000000000000:instance/00000000-0000-0000-0000-000000000000",
        "Name": "International Department",
        "ParentGroupArn": {
          "Fn::GetAtt": [
            "Root0SalesDivision",
            "UserHierarchyGroupArn",
          ],
        },
      },
      "Type": "AWS::Connect::UserHierarchyGroup",
    },
    "Root0SalesDivision1InternationalDepartment0TeamC": {
      "Properties": {
        "InstanceArn": "arn:aws:connect:us-east-1:000000000000:instance/00000000-0000-0000-0000-000000000000",
        "Name": "Team C",
        "ParentGroupArn": {
          "Fn::GetAtt": [
            "Root0SalesDivision1InternationalDepartment",
            "UserHierarchyGroupArn",
          ],
        },
      },
      "Type": "AWS::Connect::UserHierarchyGroup",
    },
    "Root0SalesDivision1InternationalDepartment1TeamD": {
      "Properties": {
        "InstanceArn": "arn:aws:connect:us-east-1:000000000000:instance/00000000-0000-0000-0000-000000000000",
        "Name": "Team D",
        "ParentGroupArn": {
          "Fn::GetAtt": [
            "Root0SalesDivision1InternationalDepartment",
            "UserHierarchyGroupArn",
          ],
        },
      },
      "Type": "AWS::Connect::UserHierarchyGroup",
    },
    "Root0SalesDivision1InternationalDepartment2TeamE": {
      "Properties": {
        "InstanceArn": "arn:aws:connect:us-east-1:000000000000:instance/00000000-0000-0000-0000-000000000000",
        "Name": "Team E",
        "ParentGroupArn": {
          "Fn::GetAtt": [
            "Root0SalesDivision1InternationalDepartment",
            "UserHierarchyGroupArn",
          ],
        },
      },
      "Type": "AWS::Connect::UserHierarchyGroup",
    },
    "Root1SupportDivision": {
      "Properties": {
        "InstanceArn": "arn:aws:connect:us-east-1:000000000000:instance/00000000-0000-0000-0000-000000000000",
        "Name": "Support Division",
      },
      "Type": "AWS::Connect::UserHierarchyGroup",
    },
    "Root1SupportDivision0TechnicalDepartment": {
      "Properties": {
        "InstanceArn": "arn:aws:connect:us-east-1:000000000000:instance/00000000-0000-0000-0000-000000000000",
        "Name": "Technical Department",
        "ParentGroupArn": {
          "Fn::GetAtt": [
            "Root1SupportDivision",
            "UserHierarchyGroupArn",
          ],
        },
      },
      "Type": "AWS::Connect::UserHierarchyGroup",
    },
    "Root1SupportDivision0TechnicalDepartment0TeamF": {
      "Properties": {
        "InstanceArn": "arn:aws:connect:us-east-1:000000000000:instance/00000000-0000-0000-0000-000000000000",
        "Name": "Team F",
        "ParentGroupArn": {
          "Fn::GetAtt": [
            "Root1SupportDivision0TechnicalDepartment",
            "UserHierarchyGroupArn",
          ],
        },
      },
      "Type": "AWS::Connect::UserHierarchyGroup",
    },
    "Root1SupportDivision1CustomerDepartment": {
      "Properties": {
        "InstanceArn": "arn:aws:connect:us-east-1:000000000000:instance/00000000-0000-0000-0000-000000000000",
        "Name": "Customer Department",
        "ParentGroupArn": {
          "Fn::GetAtt": [
            "Root1SupportDivision",
            "UserHierarchyGroupArn",
          ],
        },
      },
      "Type": "AWS::Connect::UserHierarchyGroup",
    },
    "Root1SupportDivision1CustomerDepartment0TeamG": {
      "Properties": {
        "InstanceArn": "arn:aws:connect:us-east-1:000000000000:instance/00000000-0000-0000-0000-000000000000",
        "Name": "Team G",
        "ParentGroupArn": {
          "Fn::GetAtt": [
            "Root1SupportDivision1CustomerDepartment",
            "UserHierarchyGroupArn",
          ],
        },
      },
      "Type": "AWS::Connect::UserHierarchyGroup",
    },
    "Root1SupportDivision1CustomerDepartment1TeamH": {
      "Properties": {
        "InstanceArn": "arn:aws:connect:us-east-1:000000000000:instance/00000000-0000-0000-0000-000000000000",
        "Name": "Team H",
        "ParentGroupArn": {
          "Fn::GetAtt": [
            "Root1SupportDivision1CustomerDepartment",
            "UserHierarchyGroupArn",
          ],
        },
      },
      "Type": "AWS::Connect::UserHierarchyGroup",
    },
  },
}
`;

exports[`AmazonConnectContentStack > Matches the snapshot > Not create hierarchy 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "HoursOfOperation": {
      "Properties": {
        "Config": [
          {
            "Day": "MONDAY",
            "EndTime": {
              "Hours": 17,
              "Minutes": 0,
            },
            "StartTime": {
              "Hours": 9,
              "Minutes": 0,
            },
          },
          {
            "Day": "TUESDAY",
            "EndTime": {
              "Hours": 17,
              "Minutes": 0,
            },
            "StartTime": {
              "Hours": 9,
              "Minutes": 0,
            },
          },
          {
            "Day": "WEDNESDAY",
            "EndTime": {
              "Hours": 17,
              "Minutes": 0,
            },
            "StartTime": {
              "Hours": 9,
              "Minutes": 0,
            },
          },
          {
            "Day": "THURSDAY",
            "EndTime": {
              "Hours": 17,
              "Minutes": 0,
            },
            "StartTime": {
              "Hours": 9,
              "Minutes": 0,
            },
          },
          {
            "Day": "FRIDAY",
            "EndTime": {
              "Hours": 17,
              "Minutes": 0,
            },
            "StartTime": {
              "Hours": 9,
              "Minutes": 0,
            },
          },
        ],
        "Description": "Standard business hours for operations",
        "InstanceArn": "arn:aws:connect:us-east-1:000000000000:instance/00000000-0000-0000-0000-000000000000",
        "Name": "Business Hours",
        "TimeZone": "UTC",
      },
      "Type": "AWS::Connect::HoursOfOperation",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
